### ISSUE TITLE
Onnx export, load, model check fails with segfault for onnx 1.6

### ISSUE THREAD

**Issue by dineNshine** [pytorch/pytorch#30560]
## 🐛 Bug

Exporting a module to the onnx format and then loading it with the newest onnx (1.6) fails with a segfault. The export works with onnx 1.4.1.

## To Reproduce

```
import torch
import torch.nn as nn
# import caffe2.python.onnx.backend as backend
import onnx


class Model(nn.Module):
  def __init__(self):
    super(Model, self).__init__()
  
  def forward(self, x):
    return 2 * x


device = 'cuda'

model = Model().eval().to(device)

input_data = torch.tensor([[0.0, 0.0]], device=device)
torch.onnx.export(model, input_data, 'model.onnx', verbose=True)

onnx_model = onnx.load('model.onnx')
onnx.checker.check_model(onnx_model)

# rep = backend.prepare(onnx_model, device)
# onnx_model_out = rep.run(input_data.numpy())

```

Running a forward pass with the loaded model with the caffe2 backend instead of checking also causes a segfault.

## Expected behavior

The error should be more helpful. There could be a version check and a warning if the onnx version is wrong, for example.

This might be an onnx package problem, I am not sure.

## Environment

 - PyTorch Version (e.g., 1.0): 1.3.1
 - OS (e.g., Linux): Ubuntu 16.04
 - How you installed PyTorch (`conda`, `pip`, source): pip3
 - Python version: 3.5.2
 - CUDA/cuDNN version: 10.2, 7.6
 - GPU models and configuration: 2 x GTX 1080 Ti

cc @ezyang @gchanan @zou3519 @jerryzh168 @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by mruberry** [comment]
Thanks for reporting this issue!

**Comment by bpstark** [comment]
I was trying to reproduce this issue and not having any success in reproducing.
# Environment
* PyTorch Version: 1.3.1
* OS: Ubuntu 18.04
* How you installed PyTorch: conda install pytorch=1.3.1 -c pytorch
* Python version: 3.5.4 (conda does not appear to have 3.5.2 so I got as close to your env as possible)
* CUDA/cuDNN version: 10.1, 7.6
* GPU models and configuration: 1 x GTX 1070

I installed onnx from source using the v1.6.0 tag

@pfrendl Are you exporting the model with onnx v1.4.1 then trying to load it with 1.6.0, or are you saying that if you use 1.4.1 to export and load that works, but using 1.6.0 to export and load causes a segfault?

**Comment by Wheest** [comment]
Also having a similar issue

A simple single layer network is not exported properly to Onnx 1.6.  I could get it working with 1.5.0, though had to install `protobuf-compiler` from my distribution's pacakge manager to get the `pip` package working.

## To Reproduce

I have a simple case, where I have a single layer network defined thusly:

```
class ConvNet(nn.Module):
    def __init__(self):
        super(ConvNet, self).__init__()
        self.layer1 = nn.Conv2d(in_c, num_filters, kernel_size=kdim,
                      stride=stride, padding=padding, groups=1,
                      bias=False)

    def forward(self, x):
        out = self.layer1(x)
        return out

```

When I export the model to Onnx, in the usual way with `torch.onnx.export`, all is fine.

However, when I run the Onnx model checker (`onnx.checker.check_model(onnx_model)`) to see if the model is well formed, it segfaults.

The following notebook contains the code needed to reproduce the error:
<https://gist.github.com/Wheest/73a38e7abde4eb0bbc67f666f56752bd>

The model checker should see if the model is well formed, but it does not pass.

Since a model of this small size is not a common case, is this more likely to be a PyTorch issue on how it exports to Onnx, or an issue with Onnx?

## Expected behavior

The Onnx model checker should pass

## Environment

PyTorch version: 1.3.1
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Debian GNU/Linux 10 (buster)
GCC version: (Debian 8.3.0-6) 8.3.0
CMake version: version 3.13.4

Python version: 3.7
Is CUDA available: No
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1060 6GB
Nvidia driver version: 390.116
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.17.4
[pip3] torch==1.3.1
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.4.2
[pip3] onnx==1.6.0

**Comment by andfoy** [comment]
This seems to be a regression introduced in ONNX v1.6.0. I've tried to reproduce the error with the latest ONNX in master and it cannot be reproduced.

**Comment by andfoy** [comment]
@rgommers @ezyang I think this can be closed, however, we could introduce a regression test so that we can test against ONNX versions?

**Comment by rgommers** [comment]
@andfoy just to be sure: could you also reproduce it with `1.6.0` built from source, or only with the `1.6.0` binaries? Would be good to know if it's in their release process, or it was a bug they fixed.

> we could introduce a regression test so that we can test against ONNX new versions?

Were you thinking of storing an exported model that's small enough somewhere, or is it easy to install two ONNX versions in parallel?

**Comment by andfoy** [comment]
> @andfoy just to be sure: could you also reproduce it with 1.6.0 built from source, or only with the 1.6.0 binaries? Would be good to know if it's in their release process, or it was a bug they fixed.

It seems that there was some error in their release process, as when v1.6.0 tag is used from source, the error cannot be reproduced.

> Were you thinking of storing an exported model that's small enough somewhere, or is it easy to install two ONNX versions in parallel?

Maybe we could explore the first option, however, I don't know if it is worthy at all.

**Comment by rgommers** [comment]
> It seems that there was some error in their release process, as when v1.6.0 tag is used from source, the error cannot be reproduced.

Then I think this would be good to report as a bug on the ONNX issue tracker, and see if they can fix the binaries or do a `v1.6.1` release with a fixed build. Otherwise this is going to remain a pain point for users for quite a while. Could you open an issue?

> Maybe we could explore the first option, however, I don't know if it is worthy at all.

I'm not sure, it could catch future issues quickly, but on the other hand the test belongs in ONNX itself really, and if it would start failing here the resulting segfault when running the test suite would be annoying. Let's see what others think.

**Comment by andfoy** [comment]
> Then I think this would be good to report as a bug on the ONNX issue tracker, and see if they can fix the binaries or do a v1.6.1 release with a fixed build. Otherwise this is going to remain a pain point for users for quite a while. Could you open an issue?

Sure! I'll open the issue on ONNX.

**Comment by andfoy** [comment]
@rgommers I've just opened the corresponding issue on ONNX
